internal/cli/atlas/datalakepipelines/availablesnapshots/list.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This code was autogenerated at 2023-04-27T17:56:12+01:00. Note: Manual updates are allowed, but may be overwritten.
+
+package availablesnapshots
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+var listTemplate = `ID	STATUS{{range .Results}}
+{{if .DiskBackupReplicaSet}}{{.DiskBackupReplicaSet.Id}}	{{.DiskBackupReplicaSet.Status}}{{else}}{{.DiskBackupShardedClusterSnapshot.Id}}	{{.DiskBackupShardedClusterSnapshot.Status}}{{end}}{{end}}`
+
+type ListOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	cli.ListOpts
+	store store.PipelineAvailableSnapshotsLister
+
+	pipelineName   string
+	completedAfter string
+}
+
+var ErrCompletedAfterInvalidFormat = errors.New("--completedAfter flag is in an invalid format")
+
+func (opts *ListOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func convertTime(value string) *time.Time {
+	var result *time.Time
+	if completedAfter, err := time.Parse(time.RFC3339, value); err == nil {
+		result = &completedAfter
+	}
+	return result
+}
+
+func (opts *ListOpts) validate() error {
+	if _, err := time.Parse(time.RFC3339, opts.completedAfter); opts.completedAfter != "" && err != nil {
+		return fmt.Errorf("%w: expected format 'YYYY-MM-DD' got '%s'", ErrCompletedAfterInvalidFormat, opts.completedAfter)
+	}
+	return nil
+}
```

Comment:
since I have you here, can you please confirm something I've been asking the migration project for instead of this parsing can we use https://pkg.go.dev/github.com/spf13/pflag#Duration

Reply:
This is what I get:
* `Error: invalid argument "2022-01-01" for "-d, --duration" flag: time: unknown unit "-" in duration "2022-01-01"`
* `Error: invalid argument "20220101" for "-d, --duration" flag: time: missing unit in duration "20220101"`

It works with input `10h` for instance

Reply:
yeah I think we need this merged https://github.com/spf13/pflag/pull/348 :( carry on

Reply:
agreed, that would have helped a lot